corehq/apps/app_manager/models.py

```diff
@@ -4487,7 +4538,10 @@ def make_build(self, comment=None, user_id=None, previous_version=None):
             # but check explicitly so as not to change the _id if it exists
             copy._id = copy.get_db().server.next_uuid()
 
-        copy.set_form_versions(previous_version)
+        force_new_forms = False
+        if previous_version and self.build_profiles != previous_version.build_profiles:
+            force_new_forms = True
```

Comment:
Not sure I quite understand the necessity of this. If the previous version has not changed why does it need a new version number if the only change is the build profiles?